File: src/main/java/work/lclpnet/ap2/impl/game/data/entry/SimpleDataEntry.java
package work.lclpnet.ap2.impl.game.data.entry;

import org.jetbrains.annotations.Nullable;
import work.lclpnet.ap2.api.game.data.DataEntry;
import work.lclpnet.ap2.api.game.data.SubjectRef;
import work.lclpnet.kibu.translate.TranslationService;
import work.lclpnet.kibu.translate.text.TranslatedText;

public record SimpleDataEntry<Ref extends SubjectRef>(Ref subject) implements DataEntry<Ref> {

    @Nullable
    @Override
    public TranslatedText toText(TranslationService translationService) {
        return null;
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/game/data/entry/SimpleDataEntry.kt
package work.lclpnet.ap2.game.data.entry

import work.lclpnet.ap2.api.game.data.DataEntry
import work.lclpnet.ap2.api.game.data.SubjectRef
import work.lclpnet.kibu.translate.Translations
import work.lclpnet.kibu.translate.text.TranslatedText

@JvmRecord
data class SimpleDataEntry<Ref : SubjectRef>(
    override val subject: Ref,
    val data: TranslatedText?
) : DataEntry<Ref> {

    constructor(subject: Ref) : this(subject, null)

    override fun toText(translationService: Translations): TranslatedText? {
        return data
    }

    override fun scoreEquals(other: DataEntry<Ref>): Boolean {
        return other === this
    }
}
